cocos/core/scene-graph/base-node.ts

```diff
@@ -962,7 +963,14 @@ export class BaseNode extends CCObject implements ISchedulable {
 
         const component = new constructor();
         component.node = this;
+        if (this._prefab) {
+            component.__prefab = new ComponentPrefabInfo();
+            component.__prefab.fileId = component.uuid;
+        }
```

Comment:
As node's attaching process, 

we can just trigger an Editor event here: await Editor.Message.request('scene', 'link-component', nodeUuid, assetUuid);

Reply:
This mark and below marks have been adjusted.